### PR TITLE
fix: use the actual transaction index when execute failed

### PR DESCRIPTION
**Pull request by johntaiko**

### CODE REVIEW

File: lib/src/taiko/execute.rs [hunk 1]
```diff
@@ -199,27 +201,27 @@ impl TxExecStrategy<EthereumTxEssence> for TaikoTxExecStrategy {
             logs_bloom.accrue_bloom(&receipt.payload.logs_bloom);
 
             // Add receipt and tx to tries
-            let trie_key = tx_no.to_rlp();
+            let trie_key = actual_tx_no.to_rlp();
```
Comment (smtmfft) [inline lib/src/taiko/execute.rs:204]: what is the invalid tx condition here to continue the loop? Just want to check if there is one missed.

File: primitives/src/trie/mpt.rs [hunk 2]
```diff
@@ -352,25 +370,11 @@ impl MptNode {
     ///
     /// Nibbles are half-bytes, and in the context of the MPT, they represent parts of
     /// keys.
+    #[inline]
     pub fn nibs(&self) -> Vec<u8> {
         match &self.data {
             MptNodeData::Null | MptNodeData::Branch(_) | MptNodeData::Digest(_) => vec![],
-            MptNodeData::Leaf(prefix, _) | MptNodeData::Extension(prefix, _) => {
```
Comment (smtmfft) [inline primitives/src/trie/mpt.rs:358]: Do we have bug in mpt side?? if not, maybe we can make this PR cleaner and make another PR for mpt.

File: primitives/src/block.rs [hunk 3]
```diff
@@ -61,6 +61,9 @@ pub struct Header {
     /// Shanghai update.
     #[serde(default)]
     pub withdrawals_root: Option<B256>,
+    #[rlp(skip)]
+    #[serde(skip)]
+    pub transactions: Vec<EthereumTransaction>,
```
[the remark this answers is not in the document]
Comment (dyxushuai) [reply primitives/src/block.rs:66]: For debugging

File: lib/src/taiko/execute.rs (hunk 1 above)
Comment (dyxushuai) [reply lib/src/taiko/execute.rs:204]: Because our txlist comes from l1 contract, we need to keep the same logic with tiako-geth(skip invalid tx)

Comment (johntaiko) [reply lib/src/taiko/execute.rs:204]: Because our txlist comes from l1 contract, we need to keep the same logic with tiako-geth(skip invalid tx)

File: primitives/src/trie/mpt.rs (hunk 2 above)
Comment (johntaiko) [reply primitives/src/trie/mpt.rs:358]: These codes are the newest from upstream

File: primitives/src/block.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (johntaiko) [reply primitives/src/block.rs:66]: For debugging, we can print the diff of transactions between geth and raiko

File: lib/src/taiko/execute.rs (hunk 1 above)
Comment (smtmfft) [reply lib/src/taiko/execute.rs:204]: I know, what I want to check is if you are the same as taiko-geth.

Comment (johntaiko) [reply lib/src/taiko/execute.rs:204]: Only the anchor transaction must succeed.
